samples/net/aws_iot/src/main.c

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/reboot.h>
+#include <zephyr/dfu/mcuboot.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_ctrl.h>
+#include <zephyr/net/conn_mgr_connectivity.h>
+#include <zephyr/net/conn_mgr.h>
+#include <net/aws_iot.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <hw_id.h>
+#include <modem/modem_info.h>
+
+#include "json_payload.h"
+
+/* Register log module */
+LOG_MODULE_REGISTER(aws_iot_sample, CONFIG_AWS_IOT_SAMPLE_LOG_LEVEL);
+
+/* Macros used to subscribe to specific Zephyr NET management events. */
+#define L4_EVENT_MASK (NET_EVENT_L4_CONNECTED | NET_EVENT_L4_DISCONNECTED)
+#define CONN_LAYER_EVENT_MASK (NET_EVENT_CONN_IF_FATAL_ERROR)
+
+#define MODEM_FIRMWARE_VERSION_SIZE_MAX 50
+
+/* Macro called upon a fatal error, reboots the device. */
+#define FATAL_ERROR()					\
+	LOG_ERR("Fatal error! Rebooting the device.");	\
+	LOG_PANIC();					\
+	IF_ENABLED(CONFIG_REBOOT, (sys_reboot(0)))
+
+/* Zephyr NET management event callback structures. */
+static struct net_mgmt_event_callback l4_cb;
+static struct net_mgmt_event_callback conn_cb;
+
+/* Forward declarations. */
+static void shadow_update_work_fn(struct k_work *work);
+static void connect_work_fn(struct k_work *work);
+static void aws_iot_event_handler(const struct aws_iot_evt *const evt);
+
+/* Work items used to control some aspects of the sample. */
+static K_WORK_DELAYABLE_DEFINE(shadow_update_work, shadow_update_work_fn);
+static K_WORK_DELAYABLE_DEFINE(connect_work, connect_work_fn);
+
+/* Static functions */
+
+static int app_topics_subscribe(void)
+{
+	int err;
+	static char custom_topic[75] = "my-custom-topic/example";
+	static char custom_topic_2[75] = "my-custom-topic/example_2";
```

Comment:
Should this be `const`? Also, I don't think there's a need to specify the array size, just let it be as much as really needed, according to the initializing string..

Reply:
Yes, and yes.

Reply:
Fixed